### PR TITLE
Fix data race in streamEvents.

### PR DESCRIPTION
**Pull request by fische**
While investigating an issue with builds stuck in executing, I noticed there's a potential data race in `streamEvents`. If I understand the code correctly, it should be harmless at the moment.

### CODE REVIEW

**Review by peterebden — COMMENTED** [review]
I'm not sure I see how this was a race before? The `Send` call should be synchronous in regard to the Go-side argument passed to it.

**Comment by fische** [comment]
> I'm not sure I see how this was a race before? The `Send` call should be synchronous in regard to the Go-side argument passed to it.

The issue is not with `Send` but with the assignment to `op.Name`. Since multiple routines get the same pointer to `op`, they all try to write to the same `op.Name`.

**Comment by peterebden** [comment]
> The issue is not with `Send` but with the assignment to `op.Name`. Since multiple routines get the same pointer to `op`, they all try to write to the same `op.Name`.

Ah right. I am not really certain if that line is needed - but if we want to be safe it seems like we could do it once in `process` since apparently proto types are now not safe to copy (I feel like that's a new thing?).

**Comment by fische** [comment]
> Ah right. I am not really certain if that line is needed - but if we want to be safe it seems like we could do it once in `process` since apparently proto types are now not safe to copy (I feel like that's a new thing?).

Yeah good point, I'm wondering if we wont have the same issue in the `process` function though. I'll see if this is possible.

File: mettle/api/api.go [hunk 1]
```diff
@@ -435,7 +435,12 @@ func (s *server) process(msg *pubsub.Message) {
 					recover() // Avoid any chance of panicking from a 'send on closed channel'
 				}()
 				log.Debug("Dispatching update for %s", key)
-				ch <- op
+				ch <- &longrunning.Operation{
+					Name:     op.Name,
+					Metadata: op.Metadata,
+					Done:     op.Done,
+					Result:   op.Result,
+				}
```
Comment (fische) [inline mettle/api/api.go:443]: I dont really like this, but I dont think there's any other way to copy `op` since it contains a mutex.

**Review by peterebden — APPROVED** [review]

**Pull request merged**